crates/rolldown/tests/rolldown/function/experimental/strict_execution_order/ensure_lazy_module_eval/artifacts.snap

```diff
@@ -0,0 +1,44 @@
+---
+source: crates/rolldown_testing/src/integration_test.rs
+---
+# Assets
+
+## dynamic.js
+
+```js
+import { require_dynamic } from "./dynamic2.js";
+
+export default require_dynamic();
+
+```
+## dynamic2.js
+
+```js
+// HIDDEN [rolldown:runtime]
+//#region dynamic.js
+var require_dynamic = /* @__PURE__ */ __commonJS({ "dynamic.js": ((exports, module) => {
+	module.exports = 1e3;
+	globalThis.a = 1e3;
+}) });
+
+//#endregion
+export { __toESM, require_dynamic };
+```
+## main.js
+
+```js
+import { __toESM, require_dynamic } from "./dynamic2.js";
+import { strictEqual } from "node:assert";
+
+//#region sideeffects.js
+var import_dynamic = /* @__PURE__ */ __toESM(require_dynamic());
+globalThis.a = 2;
+
+//#endregion
+//#region main.js
+const a = globalThis.a;
+strictEqual(a, 2);
+
+//#endregion
+export { a };
+```
```

Comment:
The result of esbuild, https://hyrious.me/esbuild-repl/?version=0.23.0&b=e%00entry.js%00import+%7B+strictEqual+%7D+from+%27node%3Aassert%27%3B%0Aexport+const+a+%3D+globalThis.a%3B%0Aimport+%27.%2Fdynamic.js%27%0Aimport+%27.%2Fsideeffects.js%27%0A%0AstrictEqual%28a%2C+2%29%0A&b=e%00dynamic.js%00module.exports+%3D+1000%3B%0A%0AglobalThis.a+%3D+1000%3B%0A&b=%00sideeffects.js%00%0AglobalThis.a+%3D+2%3B%0A&o=%7B%0A++treeShaking%3A+true%2C%0A++external%3A+%5B%22c%22%2C+%22a%22%2C+%22b%22%2C+%22node%3Aassert%22%5D%2C%0A%22bundle%22%3A+true%2C%0Aformat%3A+%22esm%22%2C%0Aoutdir%3A+%22dist%22%2C%0Asplitting%3A+true%0A%7D